File: api/models/thread.go
package models

import (
	"time"
)

type Thread struct {
	ID         uint32    `json:"id"`
	UserID     uint32    `json:"userID"`
	Title      string    `json:"title"`
	CategoryID uint32    `json:"categoryID"`
	CreatedAt  time.Time `json:"createdAt"`
}

File: ui/src/api/thread.js
export function loadThreads(categoryID, page, pageSize) {
  let url = new URL(process.env.API_URL + "/thread");
  let params;
  if (categoryID !== undefined) params = { categoryID, page, pageSize };
  else params = { page, pageSize };
  url.search = new URLSearchParams(params).toString();
  return fetch(url, {
    method: "GET",
  });
}

export function loadThread(id) {
  return fetch(process.env.API_URL + "/thread/" + id, {
    method: "GET",
  });
}

export function createThread(token, title, content, categoryId) {
  return fetch(process.env.API_URL + "/thread", {
    method: "POST",
    body: JSON.stringify({ token, title, content, categoryId }),
  });
}
